t/break-undo.vim

```diff
@@ -0,0 +1,125 @@
+" NB: MacVim defines several key mappings such as <D-v> by default.
+" The key mappings are defined from the core, not from any runtime file.
+" So that the key mappings are always defined even if Vim is invoked by
+" "vim -u NONE" etc.  Remove the kay mappings to ensure that there is no key
+" mappings, because some tests in this file assume such state.
+imapclear
+cmapclear
+
+let g:smartinput_break_undo = 1
+runtime! plugin/smartinput.vim
+
+call vspec#hint({'scope': 'smartinput#scope()', 'sid': 'smartinput#sid()'})
+set backspace=indent,eol,start
+filetype plugin indent on
+syntax enable
+
+describe 'g:smartinput_break_undo'
+  before
+    SaveContext
```

Comment:
Remove this.   You don't have to test s: variables.  `call smartinput#clear_rules()` is just enough to reset the whole state.